core/actions/_lib/runActionInstance.ts

```diff
@@ -107,10 +107,19 @@ const _runActionInstance = async (
 	const parsedConfig = action.config.schema.safeParse(actionInstance.config ?? {});
 
 	if (!parsedConfig.success) {
-		return {
+		const err = {
 			error: "Invalid config",
 			cause: parsedConfig.error,
 		};
+		if (args.actionInstanceArgs) {
+			// Check if the args passed can substitute for missing or invalid config
+			const argsParsedAsConfig = action.config.schema.safeParse(args.actionInstanceArgs);
+			if (!argsParsedAsConfig.success) {
+				return err;
+			}
+		} else {
+			return err;
+		}
```

Comment:
Before this change we were validating the config for every action run, even when parameters are supplied, meaning actions would fail even if all the required data was provided. Now you should be able to run an action with invalid config if you supply all the parameters it would need. This came up here because in my test, I create an email action and then run it from the pub details page, without configuring it. 

LMK if you think this is a bad change. I mostly did it because it was easier than configuring the action in my test. But I also think it's an improvement for users, since we don't actually prevent you from saving an action with invalid config, so it's easy to encounter this situation.

Reply:
I also think we should change how actions work a bit so that they're not individually responsible for implementing the config/parameters merging, which this would be part of